src/nbl/builtin/CMakeLists.txt

```diff
@@ -1,265 +1,293 @@
-  "nbl/builtin/glsl/ext/RadeonRays/ray.glsl"
-  "nbl/builtin/glsl/ext/MitsubaLoader/instance_data_struct.glsl"
-  "nbl/builtin/glsl/ext/MitsubaLoader/instance_data_descriptor.glsl"
-  "nbl/builtin/glsl/ext/MitsubaLoader/material_compiler_compatibility_impl.glsl"
-  # OIT
-  "nbl/builtin/glsl/ext/OIT/oit.glsl"
-  "nbl/builtin/glsl/ext/OIT/insert_node.glsl"
-  "nbl/builtin/glsl/ext/OIT/resolve.frag"
-  # virtual geometry
-  "nbl/builtin/glsl/virtual_geometry/descriptors.glsl"
-  "nbl/builtin/glsl/virtual_geometry/virtual_attribute.glsl"
-  "nbl/builtin/glsl/virtual_geometry/virtual_attribute_fetch.glsl"
-  # depth pyramid generator
-  "nbl/builtin/glsl/ext/DepthPyramidGenerator/common.glsl"
-  "nbl/builtin/glsl/ext/DepthPyramidGenerator/push_constants_struct_common.h"
-  "nbl/builtin/glsl/ext/DepthPyramidGenerator/depth_pyramid_generator_impl.glsl"
-  "nbl/builtin/glsl/ext/DepthPyramidGenerator/virtual_work_group.glsl"
-  # blit
-  "nbl/builtin/glsl/blit/formats_encode.glsl"
-  "nbl/builtin/glsl/blit/parameters.glsl"
-  "nbl/builtin/glsl/blit/multi_dimensional_array_addressing.glsl"
-  "nbl/builtin/glsl/blit/default_compute_common.comp"
+	"nbl/builtin/glsl/blit/default_compute_normalization.comp"
+	"nbl/builtin/glsl/blit/normalization/normalization.glsl"
+	"nbl/builtin/glsl/blit/normalization/descriptors.glsl"
+	"nbl/builtin/glsl/blit/normalization/shared_normalization.glsl"
+PARENT_SCOPE)
 
-  "nbl/builtin/glsl/blit/default_compute_blit.comp"
-  "nbl/builtin/glsl/blit/blit/blit.glsl"
-  "nbl/builtin/glsl/blit/blit/descriptors.glsl"
+function(ADD_CUSTOM_BUILTIN_RESOURCES _TARGET_NAME_ _NAMESPACE_ _CUSTOM_RESOURCES_TO_EMBED_ _OUTPUT_DIRECTORY_)
+	set(NBL_BUILTIN_HEADER_GEN_PY "${CMAKE_CURRENT_FUNCTION_LIST_DIR}/builtinHeaderGen.py")
+	set(NBL_BUILTIN_DATA_GEN_PY "${CMAKE_CURRENT_FUNCTION_LIST_DIR}/builtinDataGen.py")
 
-  "nbl/builtin/glsl/blit/default_compute_alpha_test.comp"
-  "nbl/builtin/glsl/blit/alpha_test/alpha_test.glsl"
-  "nbl/builtin/glsl/blit/alpha_test/descriptors.glsl"
+	foreach(X IN LISTS _CUSTOM_RESOURCES_TO_EMBED_)
+	list(APPEND NBL_DEPENDENCY_FILES "${NBL_ROOT_PATH}/include/${X}")
+	endforeach()
 
-  "nbl/builtin/glsl/blit/default_compute_normalization.comp"
-  "nbl/builtin/glsl/blit/normalization/normalization.glsl"
-  "nbl/builtin/glsl/blit/normalization/descriptors.glsl"
-  "nbl/builtin/glsl/blit/normalization/shared_normalization.glsl"
+	list(APPEND NBL_DEPENDENCY_FILES "${NBL_BUILTIN_HEADER_GEN_PY}")
+	list(APPEND NBL_DEPENDENCY_FILES "${NBL_BUILTIN_DATA_GEN_PY}")
 
-PARENT_SCOPE)
+	set(NBL_RESOURCES_LIST_FILE "${NBL_ROOT_PATH_BINARY}/include/nbl/builtin/resources.txt")
+
+	string(REPLACE ";" "," RESOURCES_ARGS "${_CUSTOM_RESOURCES_TO_EMBED_}")
+	file(MAKE_DIRECTORY "${NBL_ROOT_PATH_BINARY}/include/nbl/builtin")
+	file(WRITE "${NBL_RESOURCES_LIST_FILE}" "${RESOURCES_ARGS}")
+
+	set(NBL_BS_HEADER_FILENAME "builtinResources.h")
+	set(NBL_BS_DATA_SOURCE_FILENAME "builtinResourceData.cpp")
```

Comment:
want these configurable probably.

The header NEEDS to install as part of the install target (otherwise external projects can't see them)
[also this include dir needs to be a PUBLIC thing on nabla so examples can find this generated header and so can any project that includes nabla with `add_subdirectory`]